tsdb/index/postings.go

```diff
@@ -79,6 +79,45 @@ func (p *MemPostings) SortedKeys() []labels.Label {
 	return keys
 }
 
+// LabelNames returns all the unique label names, if buf is not large enough,
+// it will alloc a new one.
+func (p *MemPostings) LabelNames(buf []string) []string {
```

Comment:
Let's remove this `buf` for now in both methods as we are not using it and I don't think we will. We could add it later if this turns out to be a big overhead.